### PR TITLE
Decom example using SWE packet

### PR DESCRIPTION
**Pull request by tech3371**
# Change Summary

## Overview
Creating an example of how to unpacket CCSDS packet( aka L0 data) using Space Packet Parser and XTCE. Used SWE sample data and packet definition to demonstrate it.

## New Dependencies
None
## New Files
- Docker-compose.yml
   - Better way to build, run, and manage docker image.
   - In the file, there is one service defined. We can add more service as needed.
   - imap_processing_dev could be used as local development environment. It ensures everyone has same development environment. Please read README.docker.md for more information and benefits.
- Dockerfile
   - This was taken from our other repo and added comments and space packet parser library. Although, most of libraries in the Dockerfile are not used right now, I left it because we will need to add those library later when we need to develop CDK code.
- README.docker.md
   - Documentation on how to use docker for local development.
- swe_decom.py
   - Initial code to unpack SWE's sample data packet using SWE's packet definition.
- swe_packet_definition.xml
   - XTCE definiton to unpack SWE's CCSDS packet.

## Deleted Files
- None

## Updated Files
None

## Testing
None

### CODE REVIEW

File: in_situ/swe/swe_decom.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from pathlib import Path
+from space_packet_parser import xtcedef, parser
+import bitstring
```
Comment (greglucas) [inline in_situ/swe/swe_decom.py:3]:
```suggestion
```

File: in_situ/swe/swe_decom.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+packet_file = Path('science_block_20221116_163611Z_idle.bin')
+with packet_file.open(mode='rb') as fh:
+    binary_data = bitstring.ConstBitStream(fh)
```
Comment (greglucas) [inline in_situ/swe/swe_decom.py:7]: I think you can ignore bitstring details according to the space_packet_parser docs: https://space-packet-parser.readthedocs.io/en/latest/users.html#examples

```suggestion
packet_file = Path('science_block_20221116_163611Z_idle.bin')
with packet_file.open(mode='rb') as binary_data:
    packet_generator = my_parser.generator(binary_data)
```

```python
with packet_file.open() as binary_data:
    packet_generator = my_parser.generator(binary_data)

    for packet in packet_generator:
        # Do something with the packet data
        print(packet.header['PKT_APID'])
        print(packet.data)
```

**Review by greglucas — COMMENTED** [review]
I think we should try and clarify how we want this to work in practice. Because I see this is actually 2 separate items in the PR?

1) The packet parsing example for SWE
2) The Docker setup

For (1) I think we should make this a pip installable package, and start adding the dependencies rather than requiring the dockerfile (i.e. I don't like developing in a docker container locally if I don't have to)

For (2) I don't think this repository should be dependent on CDK, but it looks like there is a lot of AWS extras in there. I think we should look at slimming that image down or starting from a different base layer image. Possibly creating our own base image with a few requirements that we will then deploy later.

All that being said, I think (1) making it a pip installable package will solve a lot of the development issues.

File: in_situ/swe/swe_decom.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+# Initialize parser with custom XTCE definition
+my_parser = parser.PacketParser(packet_definition)
+
+# Generator object that spits out parsed Packet objects containing header and data attributes
+packet_generator = my_parser.generator(binary_data)
+
+for packet in packet_generator:
+    # Do something with the packet data
+    print(packet.header)
```
Comment (sdhoyt) [inline in_situ/swe/swe_decom.py:20]: Just for my understanding, would the next step here be to put each field into a CDF?

Comment (tech3371) [reply in_situ/swe/swe_decom.py:20]: Sorry @sdhoyt . This PR has moved to https://github.com/IMAP-Science-Operations-Center/imap_processing/pull/1. 

But to answer your questions. That's my understanding too. I envision we unpack data using packet parser. In that step we don't do any modification to data. We just unpack data.

Then we perform additional changes as needed for each instrument. Some like to leave data as it is, some like to decompress data and format it to their requirements, and some may just dump data based on flags. And write it into a CDF.

**Comment by tech3371** [comment]
This PR has moved to https://github.com/IMAP-Science-Operations-Center/imap_processing/pull/1

**Pull request closed without merging**